sklearn/utils/estimator_checks.py

```diff
@@ -340,6 +342,8 @@ def check_estimator_sparse_data(name, Estimator):
     X_csr = sparse.csr_matrix(X)
     y = (4 * rng.rand(40)).astype(np.int)
     for sparse_format in ['csr', 'csc', 'dok', 'lil', 'coo', 'dia', 'bsr']:
+        if name == 'SpectralCoclustering':
```

Comment:
Sorry I am not aware of the mathematical meaning, but if the sum over an axis is 0 then the ``row_diag``and ``col_diag`` have ``nan`` and ``inf`` which is causing the ValueError [here](https://github.com/maniteja123/scikit-learn/blob/bf459f445c43a663df565bc0aa237f57acfc7264/sklearn/cluster/bicluster.py#L170)